gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsMonitor.java

```diff
@@ -15,7 +15,14 @@
 
 import io.trino.gateway.ha.config.ProxyBackendConfiguration;
 
-public interface ClusterStatsMonitor
+public abstract class ClusterStatsMonitor
 {
-    ClusterStats monitor(ProxyBackendConfiguration backend);
+    abstract ClusterStats monitor(ProxyBackendConfiguration backend);
+
+    protected void populateClusterStats(ClusterStats.Builder stats, ProxyBackendConfiguration backend)
```

Comment:
I was thinking about it, but I was trying to avoid a dependency on the `ProxyBackendConfiguration`, but since you brought up the same point, makes me thing that that's a better approach.